### PR TITLE
Norik

### PR DESCRIPTION
**Pull request by NoroAxper**

### CODE REVIEW

File: src/cohortManager.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  }
+
+  addStudent(cohortName, newStudentFirstName, newStudentLastName) {
+    this.cohortManager.forEach((team) => {
+      for (const key in team) {
+        for (let i = 0; i < team[key].length; i++) {
+          if (
+            team[key][i].firstName === newStudentFirstName &&
+            team[key][i].lastName === newStudentLastName
+          ) {
```
Comment (ample-samples) [inline src/cohortManager.js:35]: What is the purpose of this bit? Might you be able to use a find function for this?

File: src/cohortManager.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+  removeStudent(cohortName, studentFirstName, studentLastName) {
+    let idxOfStudent
+    let idxOfCohort
+    const resultOfCohortFind = this.cohortManager.find((obj, idx) => {
+      if (obj.hasOwnProperty(cohortName)) {
+        idxOfCohort = idx
+        return obj
+      } else return false
+    })
```
Comment (ample-samples) [inline src/cohortManager.js]: Pretty cool use of .find() !

File: spec/test.spec.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
+const CohortDetainment = require('../src/cohortManager.js')
+
+describe('CohortDetainment', () => {
+  let cd
+  beforeEach(() => {
+    cd = new CohortDetainment()
```
Comment (ample-samples) [inline spec/test.spec.js:6]: Bro is this a prison manager xD

**Review by ample-samples — COMMENTED** [review]
Good work

File: spec/test.spec.js
Comment (Rebsonau) [inline spec/test.spec.js:1]: Really good work. To make it easier to follow I would suggest putting a blank line between each 'it'. That's just me being picky but nice job.

File: src/cohortManager.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    this.cohortManager = []
+    this.idCount = 1
+  }
+
+  createCohort(cohortName) {
+    if (cohortName.length < 1) {
+      throw new Error('Empty string is not valid!')
+    }
+    this.cohortManager.forEach((teamName) => {
+      for (const key in teamName) {
```
Comment (Rebsonau) [inline src/cohortManager.js:12]: Nice to see for in being utilised 👌

File: src/domain-model.md
Comment (Rebsonau) [inline src/domain-model.md:1]: Really organised here

File: src/cohortManager.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  }
+
+  allStudentsNamed(name) {
+    const arrayWithSameNameStudents = []
+    this.cohortManager.forEach((team) => {
+      for (const key in team) {
+        for (let i = 0; i < team[key].length; i++) {
+          if (
+            team[key][i].firstName === name ||
+            team[key][i].lastName === name
```
Comment (Rebsonau) [inline src/cohortManager.js:132]: You could potentially store team[key].length in a variable and then call that. Would save writing it out repeatedly.

**Review by Rebsonau — COMMENTED** [review]
Nicely done, can't pick faults within it.

File: src/cohortManager.js (hunk 1 above)
Comment (NoroAxper) [reply src/cohortManager.js:35]: potentially, but i was looking at objects within objects and arrays so could more easily map it in my head like this

**Pull request closed without merging**